lib/mof/scanner.rb

```diff
@@ -125,7 +125,7 @@ def fill_queue
 #	$stderr.puts ":string(#{scanner[1]})"
 	@q.push [:stringValue, scanner[1]]
 
-      when m = scanner.scan(%r{\w+})
+      when m = scanner.scan(%r{[\w.]+})
```

Comment:
I think that you mean `[\w\.]+`, not `[\w.]+`, because if you really meant the latter (where `.` matches any character), you could rewrite it as `.+`.